### PR TITLE
Updating project documentation for catching up with source code

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
 
 ## 3.1. Introduction
 
-The OSConfig Agent handles communication with IoT Hub and Reported Configuration/Desired Configuration (RC/DC) files. The agent is a thin client running as a daemon (Linux background service) using the IoT Client APIs from the [Azure IoT PnP C SDK](https://github.com/Azure/azure-iot-sdk-c) to connect to the IoT Hub that receives Desired Twin updates from and makes Reported Twin updates to the IoT Hub and the RC/DC files. 
-
-The agent relies on the [Azure Identity Services (AIS)](https://azure.github.io/iot-identity-service/) to obtain a module identity that allows it to connect to the IoT Hub alongside other locally present PnP agents. 
+The OSConfig agent is a thin client running as a daemon (Linux background service) and implements several adapters: a watcher for the Reported Configuration/Desired Configuration (RC/DC), a watcher for GitOps and a PnP agent for IoT Hub.
```
[the remark this answers is not in the document]
Comment (MariusNi) [reply docs/architecture.md:38]: Not going to add links for everything in there. Certain things people can simply search for.

File: docs/architecture.md [hunk 2]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
...
 
-The agent is completely decoupled from the platform and the modules. A new module can be installed, and necessary PnP interface(s) published as part of the OSConfig Model and that will be enough to make OSConfig use the respective module, without the need to recompile the agent or the platform.
+The agent is completely decoupled from the platform and the modules. A new module can be installed, and, optionally, PnP interface(s) published as part of the OSConfig Model and that will be enough to make OSConfig use the respective module, without the need to recompile the agent or the platform.
+
+## 3.2. RC/DC Watcher
+
+The RC/DC Watcher monitors the Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. The Watcher also makes MpiGetReported calls to the Management Platform and writes the reported configuration to the Reported Configuration (RC) file. 
 
-## 3.2. AIS Client
+The RC/DC files are written in MIM JSON format.
```
[the remark this answers is not in the document]
Comment (AhmedBM) [reply docs/architecture.md]: nit: add link to MIM https://github.com/Azure/azure-osconfig/blob/ahbenmes/moduletest_distro_onboard/src/modules/schema/mim.schema.json

File: docs/architecture.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MariusNi) [reply docs/architecture.md:38]: AIS and PnP C SDK are less known and also we code against them so why the links

**Review by robertschaedler3 — APPROVED** [review]

File: docs/architecture.md [hunk 3]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
...
+The  RC/DC files by default reside under /etc/osconfig/ as `/etc/osconfig/osconfig_desired.json` (the DC file) and `/etc/osconfig/osconfig_reported.json` (the RC file).
 
-The agent's  IoT Hub Client parses the Desired Twin payload received from the IoT Hub to separate individual PnP property values. The agent does not attempt to parse the property values. 
+## 3.3. GitOps Watcher
 
-## 3.4. Watcher
+The RC/DC Watcher clones a Git repository, monitors the GitOps Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. 
 
-The Watcher monitors the Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. The Watcher also makes MpiGetReported calls to the Management Platform and writes the reported configuration to the Reported Configuration (RC) file. 
+The GitOps DC file by default is named `osconfig_desired.json` and is located in the root of the repository at `/etc/osconfig/gitops/osconfig_desired.json`.
```
Comment (AhmedBM) [inline docs/architecture.md:60]: is was initially confused by this statement, consider re-wording?
"the root of the defined repository to watch"?

File: docs/architecture.md [hunk 4]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
...
 
-The pair of DC and RC files are written in MIM JSON format. The RC/DC files act as local Device Twins. 
+The GitOps DC file is written in MIM JSON format.
```
Comment (AhmedBM) [inline docs/architecture.md]: nit: consider adding link to [mim.schema.json](https://github.com/Azure/azure-osconfig/blob/ahbenmes/moduletest_distro_onboard/src/modules/schema/mim.schema.json) here

Comment (MariusNi) [reply docs/architecture.md]: Please review last commit. Refresh.

File: docs/architecture.md (hunk 3 above)
Comment (MariusNi) [reply docs/architecture.md:60]: It's not that.... but sure

File: docs/architecture.md [hunk 5]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
...
 
-When the agent starts, it receives the full Desired Twin and dispatches that to the platform for modules. From there on, incremental changes of the Desired Twin are communicated to agent, one (full or partial) property at a time. 
+When the agent starts, it receives the full Desired Twin and dispatches that to the platform. From there on, incremental changes of the Desired Twin are communicated to agent, one (full or partial) property at a time. 
+
+In the opposite direction, the OSConfig agent periodically updates the Reported Twin with one property value at a time, reading via the platform from the modules. 
+
+## 3.7. MPI Client
 
-In the opposite direction, OSConfig agent periodically updates the Reported Twin with one property value at a time, reading via the platform from the modules. 
+The agent implements an MPI Client and it uses it to make Management Platform Interface (MPI) calls to the platform as IPC calls over HTTP and the Unix Domain Sockets (UDS).
```
Comment (AhmedBM) [inline docs/architecture.md:86]: consider calling out "REST" here instead of "IPC over HTTP"

**Review by robertschaedler3 — APPROVED** [review]

**Review by AhmedBM — APPROVED** [review]
👌

File: docs/architecture.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
-Azure Device OS Configuration (OSConfig) - North Star Architecture
-==================================================================
+Azure OSConfig - North Star Architecture
+========================================
 Author: [MariusNi](https://github.com/MariusNi)
 
 # 1. Introduction
 
-Azure Device OS Configuration (OSConfig) is a modular configuration stack for�Linux Edge devices. OSConfig supports multi-authority device management over Azure and Azure Portal/CLI (via Azure PnP, IoT Hub, Azure Policy), GitOps, as well as local management (such as from Out Of Box Experience, OOBE).
+Azure OSConfig is a modular security configuration stack for�Linux Edge devices. OSConfig supports multi-authority device management over Azure and Azure Portal/CLI, GitOps, as well as local management.
```
Comment (simonjaeger) [inline docs/architecture.md:7]: Some character rendering strangely: 
image

File: docs/architecture.md [hunk 7]
```diff
@@ -41,62 +35,70 @@ This diagram shows the current North Star architecture of OSConfig. In this diag
...
+The RC/DC Watcher monitors the Desired Configuration (DC) file and acts on detected file changes by making MpiSetDesired calls to the Management Platform. The Watcher also makes MpiGetReported calls to the Management Platform and writes the reported configuration to the Reported Configuration (RC) file. 
 
-## 3.2. AIS Client
+The RC/DC files are written in JSON and follow the [MIM schema](../src/modules/schema/mim.schema.json).
 
-The agent  uses the PnP C SDK's HTTP helper APIs to make GET and POST requests to the [Azure Identity Services (AIS)](https://azure.github.io/iot-identity-service/) (identity, key, certificates) to build the device/module identity to (re)connect to the IoT Hub with. 
+To protect against unauthorized access, the RC/DC files are restricted to root user read and write access only.
 
-## 3.3. IoT Hub Client
+The  RC/DC files by default reside under `/etc/osconfig/` as `/etc/osconfig/osconfig_desired.json` (the DC file) and `/etc/osconfig/osconfig_reported.json` (the RC file).
```
Comment (simonjaeger) [inline docs/architecture.md:54]: nit: Double space.

**Review by simonjaeger — APPROVED** [review]

File: docs/architecture.md (hunk 7 above)
Comment (MariusNi) [reply docs/architecture.md:54]: Nice catch!

File: docs/architecture.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MariusNi) [reply docs/architecture.md]: Added couple, thanks

**Review by robertschaedler3 — APPROVED** [review]

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**